### PR TITLE
Add generate binary tree from inorder preorder

### PR DESCRIPTION
**Pull request by omsitapara23**
(Put an `X` inside the `[ ]` to denote check mark `[X]`.)

Here, i have added the code for generating binary tree given the inorder and preorder traversals of the tree.

- [X ] **If creating a new file :**
  - [X ] added links to it in the README files ?
  - [ ] included tests with it ?
  - [ X] added description (overview of algorithm, time and space complexity, and possible edge case) in docstrings ?

- [ ] **if done some changes :**
  - [ ] wrote short description in the PR explaining what the changes do ?
  - [ ] Fixes #[issue number] if related to any issue

- [ ] **other**

### CODE REVIEW

File: algorithms/tree/binary_tree_from_preorder_inorder.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+class TreeNode( object ):
+    def __init__( self, x ):
+        self.val = x
+        self.left = None
+        self.right = None
+
+
+def generate_tree( inorder, preorder ):
+
+    if len( preorder ) <= 0 or len(inorder) <= 0:
```
Comment (goswami-rahul) [inline algorithms/tree/binary_tree_from_preorder_inorder.py]: We don't need `<=`. Just checking
```python
if not inorder:
    return None
```
will suffice.

File: algorithms/tree/binary_tree_from_preorder_inorder.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        return None
+
+    curr_node = TreeNode( preorder[0] )
+    index = inorder.index( preorder[0] )
```
Comment (goswami-rahul) [inline algorithms/tree/binary_tree_from_preorder_inorder.py]: don't leave spaces inside brackets

**Comment by keon** [comment]
Closing as stale — the repo was completely restructured in v1.0.0. Thank you for the contribution\!
